gammapy/cube/fit.py

```diff
@@ -41,28 +54,64 @@ def __init__(
         psf=None,
         edisp=None,
         background_model=None,
+        likelihood="cash",
+        evaluation_mode="local",
     ):
         if mask is not None and mask.data.dtype != np.dtype("bool"):
             raise ValueError("mask data must have dtype bool")
 
+        if isinstance(model, SkyModel):
+            model = SkyModels([model])
+
         self.model = model
         self.counts = counts
         self.exposure = exposure
         self.mask = mask
         self.psf = psf
         self.edisp = edisp
         self.background_model = background_model
-        if background_model:
-            self.parameters = Parameters(
-                self.model.parameters.parameters +
-                self.background_model.parameters.parameters
+
+        if likelihood == "cash":
+            self._stat = cash
+        elif likelihood == "cstat":
+            self._stat = cstat
+        else:
+            raise ValueError(
+                "Not a valid fit statistic. Choose between 'cash' and 'cstat'."
             )
+
+        if evaluation_mode == "local":
+            evaluators = []
+
+            for component in self.model.skymodels:
+                evaluator = MapEvaluator(component)
+                evaluators.append(evaluator)
+
+            self._evaluators = evaluators
+        elif evaluation_mode == "global":
+            evaluator = MapEvaluator(self.model, self.exposure, self.psf, self.edisp)
+            self._evaluators = [evaluator]
         else:
-            self.parameters = Parameters(self.model.parameters.parameters)
+            raise ValueError("Not a valid model evaluation mode. Choose between 'cutout' and 'global'")
 
-        self.evaluator = MapEvaluator(
-            model=self.model, exposure=exposure, psf=self.psf, edisp=self.edisp
-        )
+    @lazyproperty
+    def parameters(self):
+        """List of parameters (`~gammapy.utils.fitting.Parameters`)"""
+        if self.background_model:
+            parameters = Parameters(
+                self.model.parameters.parameters
+                + self.background_model.parameters.parameters
+            )
+        else:
+            parameters = Parameters(self.model.parameters.parameters)
+        return parameters
+
+    @property
+    def _geom(self):
+        if self.counts is not None:
```

Comment:
What is the use case of `self.counts = None`? I guess simulation, right?

Reply:
Yes, I think it's fair use case to instantiate the `MapDataset` without a counts map and then call `.npred()` to compute the expected counts from the model, randomise it and use it as counts data:
```python3
dataset = MapDataset(counts=None, ...)
npred = dataset.npred()
npred.data = np.random.poisson(npred.data)
datasets.counts = npred 
```
Shall we maybe implement a `.counts` setter? To make it safer? E.g. check whether counts is in the correct geometry...